deploy/ocpbugs-48519/10-CronJob-4-17.yaml

```diff
@@ -0,0 +1,76 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: pki-operator-annotator-4-17
+  namespace: openshift-pki-operator-annotator
+  annotations:
+    kubernetes.io/description: "Patches 4.17 ManifestWork to fixed pki-operator version https://issues.redhat.com/browse/OSD-27987"
+spec:
+  schedule: "*/5 * * * *" # Every five minutes
+  concurrencyPolicy: Replace
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          name: pki-operator-annotator
+          namespace: openshift-pki-operator-annotator
+          annotations:
+            kubernetes.io/description: "Patches 4.17 ManifestWork to fixed pki-operator version https://issues.redhat.com/browse/OSD-27987"
+        spec:
+          affinity:
+            nodeAffinity:
+              preferredDuringSchedulingIgnoredDuringExecution:
+                - preference:
+                    matchExpressions:
+                      - key: node-role.kubernetes.io/infra
+                        operator: Exists
+                  weight: 1
+          tolerations:
+            - effect: NoSchedule
+              key: node-role.kubernetes.io/infra
+              operator: Exists
+          containers:
+            - name: pki-operator-annotator
+              image: image-registry.openshift-image-registry.svc:5000/openshift/tools:latest
+              imagePullPolicy: Always
+              securityContext:
+                allowPrivilegeEscalation: false
+                capabilities:
+                  drop:
+                    - ALL
+                runAsNonRoot: true
+              env:
+                - name: IMAGE
+                  # oc adm release info --image-for hypershift quay.io/openshift-release-dev/ocp-release:4.17.15-multi
+                  value: "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:1ac8880940e256abb01cf8d8276beeb53db37572ab7369a752de5af244bc24e4"
+                - name: MAJOR_MINOR_VER
+                  value: "4.17"
```

Comment:
We'll likely want to have this down to z versions, e.g. this should apply until 4.17.14, as 4.17.15 will have the fix.

Reply:
Following the pattern we established when pinning CAPA, we opted to just pin all Y streams to the latest Z that has the fix.

To do what you're proposing would require us to find <=4.17.14 and then pin correct? I'll see if I can come up with an easy way to do that in bash/jq

Reply:
Thanks for looking into it! Without this, we'd be effectively downgrading the version for everything that comes after and might run into some future issues.

Reply:
This is now handled by a wrapper python script, which includes some unit tests to handle a couple scenarios.

Reply:
Updated script to account for removing the annotation when it is no longer needed.